pkg/splunk/client/azure_blob_rest_service.go

```diff
@@ -0,0 +1,159 @@
+package client
+
+import (
+	"context"
+)
+
+// Get the list of apps from a given
+// 1. storageaccount URL
+// 2. container name (container in azure is same as bucket in aws)
+// 3. path with the container name
+// 4. Max number of apps to return in the call
+// 5. Config object
+// 	  Timeout - time in ms that a http rest call should wait before terminating
+//    MaxRetries - number of retries for the rest call if it fails for first time.
+//    http transport related parameters
+// Output:
+//  ListBlobOutput
+//     RemoteDataClientResponse
+//     ResponseNextMarker  if total apps in the bucket were more that max number of apps requested
+
+// Download a given app with it's full path
+// 1. URL of the app to be downloaded with full path including app name
+// 2. Max number of bytes to return
+// 3. Starting byte
+// 5. Config object
+// 	  Timeout - time in ms that a http rest call should wait before terminating
+//    MaxRetries - number of retries for the rest call if it fails for first time.
+//    http transport related parameters
+// Output:
+//  DownloadBlobOutput
+//    Response Body
+//    Response Code
+//    Total Number of bytes returned
+
+type azureRestService interface {
```

Comment:
Is this a mimic of the DownloadApp and ListApps in azureblobclient.go?

Reply:
AzureBlobClient (and also similar clients like AWSS3Client, and MinioClient) provide implementation to the methods GetAppsList and DownloadApp 
```
// RemoteDataClient is an interface to provide
// listing and downloading of apps from remote data storage
//
type RemoteDataClient interface {

	// Get the list of Apps
	GetAppsList(context.Context) (RemoteDataListResponse, error)

	// Download a given app as per the inputs provided in the `RemoteDataClientRequest`
	DownloadApp(context.Context, RemoteDataDownloadRequest) (bool /* return pass/fail */, error)
}
'''

These clients will eventually use a third-party data provider. For AzureBlobClient, we are going to use our own small package (currently there is only one file azure_blob_rest_service.go in this package) to list and download apps vis Azure Rest Calls. The `DownloadApp` and `ListApps` methods will be part of this package.

Reply:
@kumarajeet Do we need a separate package for it i.e we could have only helper functions for http operations in azure_blob_rest_service.go while keeping the list and download Apis in the azurebloblclient.go?

Reply:
@akondur the simplest solution will be to keep the rest helper functions in the same file.
However, the way currently the Mockxxxxclients are working (some cyclic package inclusions constraints), we may have to keep the azure_blob_rest_service.go in its own package.